App/xcbuddykit/Sources/Generator/ProjectGroups.swift

```diff
@@ -81,85 +58,48 @@ class ProjectGroups {
         }
     }
 
-    /// Returns a group that should contain all the configurations of a given target.
-    ///
-    /// - Parameter target: target name.
-    /// - Returns: group to be used.
-    /// - Throws: an error if the creation fails.
-    func targetConfigurations(_ target: String) throws -> PBXGroup {
-        if let group = configurations.group(named: target) {
-            return group
-        } else {
-            return try configurations.addGroup(named: target, options: .withoutFolder).last!
-        }
-    }
-
-    /// Returns a group that should contain all the configurations of a project.
-    ///
-    /// - Returns: the group to be used.
-    /// - Throws: an error if the creation fails.
-    func projectConfigurations() throws -> PBXGroup {
-        if let group = configurations.group(named: "Project") {
-            return group
-        } else {
-            return try configurations.addGroup(named: "Project", options: .withoutFolder).last!
-        }
-    }
-
     /// Generates all the Xcode project base groups and returns an instance of ProjectGroups
     /// that contains a reference to them.
     ///
     /// - Parameters:
     ///   - project: project spec.
-    ///   - pbxproj: PBXProj instance.
+    ///   - objects: project objects.
     ///   - sourceRootPath: path to the folder where the Xcode project is getting created.
     /// - Returns: project groups instance.
     static func generate(project: Project,
-                         pbxproj: PBXProj,
+                         objects: PBXObjects,
                          sourceRootPath: AbsolutePath) -> ProjectGroups {
         /// Main
         let mainGroup = PBXGroup(children: [],
                                  sourceTree: .group,
                                  path: project.path.relative(to: sourceRootPath).asString)
-        pbxproj.objects.addObject(mainGroup)
-
-        /// Targets
-        let targetsGroup = PBXGroup(children: [], sourceTree: .group, name: "Targets")
-        let targetsGroupReference = pbxproj.objects.addObject(targetsGroup)
-        mainGroup.children.append(targetsGroupReference)
-
-        /// Configurations
-        let configurationsGroup = PBXGroup(children: [], sourceTree: .group, name: "Configurations")
-        let configurationsGroupReference = pbxproj.objects.addObject(configurationsGroup)
-        mainGroup.children.append(configurationsGroupReference)
+        objects.addObject(mainGroup)
 
-        /// Support
-        let supportGroup = PBXGroup(children: [], sourceTree: .group, name: "Support")
-        let supportGroupReference = pbxproj.objects.addObject(supportGroup)
-        mainGroup.children.append(supportGroupReference)
+        /// Project
+        let projectGroup = PBXGroup(children: [], sourceTree: .group, name: "Project")
```

Comment:
What do you think about having all the project files inside a root group called "Project"? This is how the project would look:

![image](https://user-images.githubusercontent.com/663605/40107031-87f40774-58f7-11e8-9638-c256db0406e4.png)

- **Frameworks:** Targets frameworks.
- **Products:** Targets products.
- **ProjectDescription:** Manifest files.
- **Project:** Files referenced by the manifests.